lib/matplotlib/axis.py

```diff
@@ -1579,10 +1579,13 @@ def set_major_locator(self, locator):
         locator : ~matplotlib.ticker.Locator
         """
         if not isinstance(locator, mticker.Locator):
-            raise TypeError("formatter argument should be instance of "
+            raise TypeError("locator argument should be instance of "
                             "matplotlib.ticker.Locator")
         self.isDefault_majloc = False
         self.major.locator = locator
+        if (hasattr(self.get_major_formatter(), "_locator") and
+            hasattr(self.get_major_formatter(), "_tz")):
+            self.get_major_formatter()._locator = locator
```

Comment:
Should this be `.set_locator(locator)` instead of `._locator`?

Reply:
Good point. Would you favour making `locator` a public property?

Reply:
I don't really now, I was tripped up earlier in my stem PR by accessing something using a private attribute instead of its setter though!

Reply:
Actually, I don't think this should happen in `axis.py` at all, but should all happen in `dates.py`, probably in `AutoDateLocator.__call__()` where it can check if the locator on the `self.axis` has changed.  I'll actually block on this, but am willing to be argued with if its really not possible any other way...